src/linear_algebra/affine_space.lean

```diff
@@ -100,6 +102,183 @@ begin
   apply set.mem_of_mem_of_subset hp1p2 hp1p2s
 end
 
+section combination
+
+variables {k} {ι : Type*} [fintype ι]
+
+/-- A weighted sum of the results of subtracting a base point from the
+given points.  The main cases of interest are where the sum of the
+weights is 0, in which case the sum is independent of the choice of
+base point, and where the sum of the weights is 1, in which case the
+sum added to the base point is independent of the choice of base
+point. -/
+def weighted_vsub_of_point (w : ι → k) (p : ι → P) (b : P) : V := ∑ i, w i • (p i -ᵥ b)
```

Comment:
Please add a `linear_map` version for the function `w ↦ weighted_vsub_of_point w p b` and `affine_map` version for `(p, b) ↦ weighted_vsub_of_point w p b`.

Reply:
Added those versions, along with the `add_torsor` instance for Pi types required for the `affine_map` version.